core/finality-grandpa/src/tests.rs

```diff
@@ -984,6 +985,7 @@ fn allows_reimporting_change_blocks() {
 			bad_justification: false,
 			needs_finality_proof: false,
 			is_new_best: true,
+			header_only: false,
```

Comment:
why is any of this exposed to the consensus engine (`header_only`, `allow_missing_state`), which only cares about headers?

Reply:
Same reasons as #3623. Fixing the pipeline is out of scope of this PR.